lib/IO/Socket/Async/SSL.pm6

```diff
@@ -175,10 +199,34 @@ class IO::Socket::Async::SSL {
         OpenSSL::Ctx::SSL_CTX_new($method)
     }
 
+    sub alpn-selector($ssl, $out, $outlen, $in, $inlen, $arg) {
+        my $buf = Buf.new;
+        for (0...$inlen-1) {
+            $buf.push: $in[$_];
+        }
+        my $protos = parse-protocol-list($buf, $inlen);
+        my $result;
+
+        if $server-protocols ~~ Callable {
+            $result = $server-protocols($protos);
+        } else {
+            return 3 if $server-protocols.elems == 0; # SSL_TLSEXT_ERR_NOACK
```

Comment:
Let's define a `my constant SSL_TLSEXT_ERR_NOACK = 0` and similar for the one below.